numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -251,6 +254,31 @@ def compile(src, name, cc, ltoir=False):
     nvrtc = NVRTC()
     program = nvrtc.create_program(src, name)
 
+    version = nvrtc.get_version()
+    if version < (11, 0):
+        # Maximum supported target architecture for 11.x is cc 9.0 in CUDA 11.8
+        raise RuntimeError(
+            "Unsupported CUDA version. CUDA 11.0 or higher is required."
+        )
+    else:
+        min_cc, max_cc = CTK_SUPPORTED[version]
```

Comment:
Don't assume that `version` is in `CTK_SUPPORTED`, or else new CUDA versions won't be testable without raising a `KeyError`. We should use `CTK_SUPPORTED.get(version, None)` or `try: ... except KeyError: ...`. We should use the "latest CUDA" key if no exact match is found, perhaps with an additional warning.

Reply:
> We should use the "latest CUDA" key if no exact match is found, perhaps with an additional warning.

This feels more robust to me. Ideally we should have all CTK documented here, or is that a too much a ask?

Reply:
> Ideally we should have all supported CTK documented here, or is that a too much a ask?

I think all the supported CTK versions are already in the dict? Maybe I am not understanding correctly.